app/main/views/send.py

```diff
@@ -262,6 +264,31 @@ def set_sender(service_id, template_id):
     )
 
 
+def remove_notify_from_sender_options(sender_details):
+    # Remove US Notify/Notify.gov from users list of sender
+    # options during message send flow
+    sender_details = [
+        sender for sender in sender_details if verify_sender_options(sender)
+    ]
+
+    return sender_details
+
+
+def verify_sender_options(sender):
+    if sender.get("sms_sender") in ["Notify.gov", "US Notify"] and sender["is_default"]:
+        return True
+    if (
+        sender.get("sms_sender") not in ["Notify.gov", "US Notify"]
+        and not sender["is_default"]
+    ):
+        return True
+    if (
+        sender.get("sms_sender") not in ["Notify.gov", "US Notify"]
+        and sender["is_default"]
+    ):
+        return True
```

Comment:
This is a huge help seeing the breakdown in this separate method, nicely done! 🎉 

There are a couple more things, though:

- These two conditional checks are actually checking for the same thing - it doesn't matter if the sender `is_default` or not, just that they're not in the list of `["Notify.gov", "US Notify"]`! 🙂 
- We need to make sure there's a fallback return value if none of the checks actually pass, which would be `False`

We could probably simplify things and take care of the return value like so:

```suggestion
    if sender.get("sms_sender") not in ["Notify.gov", "US Notify"]:
        return True
    
    return False
```

Reply:
That was completely correct! The additional logic wasn't needed. Verified it in the message send flow. Thank you for the advice and explaining that for me.